worker/lease/store.go

```diff
@@ -30,6 +30,9 @@ type StoreConfig struct {
 
 	// Logger is used to emit store-specific diagnostics.
 	Logger StoreLogger
+
+	// Ctx is the context intended for use when executing database operations.
+	Ctx context.Context
```

Comment:
From https://pkg.go.dev/context#pkg-overview

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

I know why we do this, I'm just wondering in the back of my mind if we should have a type that can give you contexts, but knows how to cancel them correctly. Like a tomb, but a context provider or something.